packages/client/hmi-client/src/workflow/ops/model-config/tera-model-config.vue

```diff
@@ -684,10 +685,10 @@ const updateConfigInitial = (inits: Initial[]) => {
 
 const updateConfigFromModel = (inputModel: Model) => {
 	if (isPetriNet.value || isStockFlow.value) {
-		knobs.value.initials = inputModel.semantics?.ode.initials ?? [];
-		knobs.value.parameters = inputModel.semantics?.ode.parameters ?? [];
+		knobs.value.initials = getModelInitials(inputModel) ?? [];
+		knobs.value.parameters = getModelParameters(inputModel) ?? [];
 	} else if (isRegNet.value) {
-		knobs.value.parameters = inputModel.model?.parameters ?? [];
+		knobs.value.parameters = getModelParameters(inputModel) ?? [];
```

Comment:
It seems like your getModelparameters and getModelInitials will not return nulls anymore, so we can probably clean up the ?? mullish coalesces

Reply:
ah yes, ill change this.  

To note...I was saving this thought for Monday but i was thinking of just saving just a model configuration object to the state rather than (parameters, initials, sources, units, etc...) to make all this easier.  I didnt want to touch this file too much